src/libPMacc/include/math/complex/Complex.tpp

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright 2013-2015 Heiko Burau, Rene Widera, Richard Pausch, Alexander Debus
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <cmath>
+#include "math/Complex.hpp"
+#include "algorithms/math.hpp"
+#include "algorithms/TypeCast.hpp"
+ 
+namespace PMacc
+{
+namespace algorithms
+{
+namespace math
+{
+
+/*  Set primary template and subsequent specialization for returning a complex number by using Euler's formula. */
+
+template<typename Type>
+struct Euler;
+
+template<typename Type>
+HDINLINE typename Euler< Type >::result euler(const Type& magnitude, const Type& phase)
+{
+    return Euler< Type > ()(magnitude, phase);
+}
+
+template<typename Type>
+HDINLINE typename Euler< Type >::result euler(const Type& magnitude, const Type& sinValue, const Type& cosValue)
+{
+    return Euler< Type > ()(magnitude, sinValue, cosValue);
+}
+
+template<typename Type>
+struct Euler
+{
+    typedef typename ::PMacc::math::Complex<Type> result;
```

Comment:
Is `::PMacc::math::` defined? I think it should be `::PMacc::algotithms::math::`

If `PMacc::math` is not defined please check your full file.